test/unit/math/prim/arr/meta/is_constant_struct_test.cpp

```diff
@@ -5,7 +5,14 @@
 TEST(MetaTraits, isConstantStruct) {
   using stan::is_constant_struct;
   using std::vector;
-  EXPECT_TRUE(is_constant_struct<vector<double> >::value);
-  EXPECT_TRUE(is_constant_struct<vector<vector<double> > >::value);
-  EXPECT_TRUE(is_constant_struct<vector<vector<vector<double> > > >::value);
+  EXPECT_TRUE(is_constant_struct<vector<double>>::value);
+  EXPECT_TRUE(is_constant_struct<vector<vector<double>>>::value);
+  EXPECT_TRUE(is_constant_struct<vector<vector<vector<double>>>>::value);
+  bool temp = is_constant_struct<vector<double>, vector<double>,
+                                 vector<double>>::value;
```

Comment:
Shouldn't we also have a test for

I'd want to see tests for the two edge cases, one where there's no type parameters and one where there's no structure.

```
is_constant_struct<>
is_constant_struct<double>
```

The first should be `true` as that's the unit for conjunction (or all/universal quantification).  

I'm not sure about the second---is this called `struct` because the `<double>` version should fail?

Reply:
`is_constant_struct<double>` is true. This one was already tested in the scal/meta along with `is_constant_struct<float>` and such. There isnt really a need to have `is_constant` and `is_constant_struct` separate. Will try and merged the two in this PR.

`is_constant_struct<>` is true, will add to the scal tests.